Tests for the test suite

m-mohr opened the pull request:
See the [README](https://github.com/Open-EO/openeo-processes/blob/add-tests/tests/README.md) for details.

This PR includes the changes from PR #490.

soxofaan commented:
I have the impression a lot of changes in the diff come from PR #490 
Is that intended?

m-mohr commented:
Yes:

> This PR includes the changes from PR https://github.com/Open-EO/openeo-processes/pull/490.

soxofaan commented:
ah ok, I see indeed

that (and the PR size) make it quite challenging to review. How do you see this being moved forward? focus on #468, or first get #490 merged?

m-mohr commented:
I think I'd try to get #490 merged first, I'll try to address your comments now.

soxofaan commented on tests/vector_to_random_points.json5:
```diff
@@ -0,0 +1,6 @@
+{
+  "id": "vector_reproject",
```
```suggestion
  "id": "vector_to_random_points",
```

soxofaan commented on tests/vector_to_regular_points.json5:
```diff
@@ -0,0 +1,6 @@
+{
+  "id": "vector_reproject",
```
```suggestion
  "id": "vector_to_regular_points",
```

soxofaan reviewed (commented):
found these looking in the test suite logs :stuck_out_tongue:

m-mohr commented:
PR has been updated to the latest draft version.
Also, added checks for the actual test files to the CI.
Removed the Changes from #490 to make it easier to review, but the tests are still written in a way as if #490 would've been merged already.

soxofaan reviewed (commented):
This PR is too large to give it a full in-depth review, but here are some random notes.

I guess  it's more important to get this merged as "good enough" so that implementations can start integrate it, than to reach perfection.

m-mohr commented on apply_kernel.json:
```diff
@@ -1,7 +1,7 @@
     "id": "apply_kernel",
     "summary": "Apply a spatial convolution with a kernel",
-    "description": "Applies a 2D convolution (i.e. a focal operation with a weighted kernel) on the horizontal spatial dimensions (axes `x` and `y`) of a raster data cube.\n\nEach value in the kernel is multiplied with the corresponding pixel value and all products are summed up afterwards. The sum is then multiplied with the factor.\n\nThe process can't handle non-numerical or infinite numerical ...
+    "description": "Applies a 2D convolution (i.e. a focal operation with a weighted kernel) on the horizontal spatial dimensions (axes `x` and `y`) of a raster data cube.\n\nEach value in the kernel is multiplied with the corresponding pixel value and all products are summed up afterwards. The sum is then multiplied with the factor.\n\nThe process can't handle non-numerical or infinite numerical ...
```
Well, they were spotted when creating the tests. They should've been in #490, but it seems I didn't catch them all.
